gradient/utils.py

```diff
@@ -32,4 +36,58 @@ def status_code_to_error_obj(status_code):
     message = 'unknown'
     if status_code in requests.status_codes._codes:
         message = requests.status_codes._codes[status_code][0]
-    return { 'error': True, 'message': message, 'status': status_code }
\ No newline at end of file
+    return { 'error': True, 'message': message, 'status': status_code }
+
+
+class PathParser(object):
+    LOCAL_DIR = 0
+    LOCAL_FILE = 1
+    GIT_URL = 2
+    S3_URL = 3
+
+    @classmethod
+    def parse_path(cls, path):
+        if cls.is_local_dir(path):
+            return cls.LOCAL_DIR
+
+        if cls.is_local_zip_file(path):
+            return cls.LOCAL_FILE
+
+        if cls.is_git_url(path):
+            return cls.GIT_URL
+
+        if cls.is_s3_url(path):
+            return cls.S3_URL
+
+        raise exceptions.WrongPathError("Given path is neither local path, nor valid URL")
+
+    @staticmethod
+    def is_local_dir(path):
+        return os.path.exists(path) and os.path.isdir(path)
+
+    @staticmethod
+    def is_local_zip_file(path):
+        return os.path.exists(path) and os.path.isfile(path) and path.endswith(".zip")
+
+    @staticmethod
+    def is_git_url(path):
+        return not os.path.exists(path) and path.endswith(".git") or path.lower().startswith("git:")
+
+    @staticmethod
+    def is_s3_url(path):
+        return not os.path.exists(path) and path.lower().startswith("s3:")
+
+
+def validate_workspace_input(input_data):
+    workspace_url = input_data.get('workspaceUrl')
+    workspace_path = input_data.get('workspace')
+    workspace_archive = input_data.get('workspaceArchive')
+
+    if (workspace_archive and workspace_path) \
+            or (workspace_archive and workspace_url) \
+            or (workspace_path and workspace_url):
+        raise click.UsageError("Use either:\n\t--workspace https://path.to/git/repository.git - to point repository URL"
+                               "\n\t--workspace /path/to/local/directory - to point on project directory"
+                               "\n\t--workspace /path/to/local/archive.zip - to point on project .zip archive"
```

Comment:
more naturally this would be something like `to point to a .zip archive` or `to use a local archive as your workspace` :)

Reply:
changed in PS-10553

Reply:
thanks @BartoszCki!